src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -121,16 +121,15 @@ def do_resolve_with_resolved_children(
             # overwrite the SnowflakeCreateTable resolving, because the child
             # attribute will be pulled directly from the cache
             resolved_child = resolved_children[logical_plan.children[0]]
-            # when the plan is for SnowflakeCreateTable, there must be a snowflake_create_table_plan_info
-            # associated with the current query generator.
-            # TODO: this check need to be relaxed when large query breakdown with temp table
-            #       is implemented, because the child attributes are not necessary to create
-            #       the temp table
-            assert self._snowflake_create_table_plan_info is not None
-            assert (
-                self._snowflake_create_table_plan_info.table_name
+            child_attributes = None
+            if (
+                self._snowflake_create_table_plan_info
+                and self._snowflake_create_table_plan_info.table_name
```

Comment:
also since we do that now, let's also update the create_query_generator function here to be more clear, we only need to extract the attribute when the creator_source is not large query breakdown. Even though we don't expect we will have actual usage of creating query generator when SnowflakeCreateTable node is coming from large query breakdown, this is more for code completness.

Reply:
@sfc-gh-aalam how about this one? if we do not do this one, the create_query_compiler will try to extract the child_atrribute out of the child plan

Reply:
sorry, what i mean is we can do 
```
if (logical_plan.creation_source != TableCreationSource.LARGE_QUERY_BREAKDOWN):
     assert self._snowflake_create_table_plan_info is not None
     assert (self._snowflake_create_table_plan_info.table_name==logical_plan.table_name)
     child_attributes = self._snowflake_create_table_plan_info.child_attributes
```

so that it explicitly errors out at the line where it needs the child attribute to be more clear. otherwise, i believe it will error out inside the save_as_table function with the check today